### PR TITLE
:warning: run BMO ironic deployment as non-root

### PR DESCRIPTION
**Pull request by tuminoid**
BMO ironic has no reason to run as root. Make it run as "ironic" user.
    
dnsmasq requires elevated capabiities. k8s is missing the feature of ambient capabilities, so it requires us to setcap the binaries with expected capabilities and container must be running with "allowPrivilegeEscalation: true" in the manifest to allow elevation.
    
Read the ambient capabilities KEP for more details: https://github.com/kubernetes/enhancements/blob/master/keps/sig-security/2763-ambient-capabilities/README.md
    
Add securityContext to BMO deployment manifest and keepalived component, with correct UIDs and GIDs. This is important to be able to share files via /shared.
    
Modify keepalived image to run as ironic user, which we use the same UID and GID as the ironic-image.
    
This commit requires ironic-image with PR https://github.com/metal3-io/ironic-image/pull/410 to be merged to work.

### CODE REVIEW

**Comment by tuminoid** [comment]
/cc @elfosardo @dtantsur

This is still WIP, but I'd appreciate early comments.

**Comment by tuminoid** [comment]
> /test-centos-e2e-integration-main

As expected, this fails. This must run on top of Ironic image change.

**Comment by tuminoid** [comment]
> The title is misleading: you're not updating just BMO. Let's maybe start with BMO itself since it is an easy win? And finish with dnsmasq which is the hardest.

All the technical issues are solved, I just keep to update the pidfile location, and as discussed in Ironic-image PR, I will reuse existing users instead of nonroot and update the UID in this one.  Combined BMO and Ironic-image test is running at https://github.com/metal3-io/metal3-dev-env/pull/1172 and were passing already.

I will fix the commit message, title and description of the PR when I get those sorted out, and remove the WIP when its ready for review and merge.

**Comment by tuminoid** [comment]
/hold

No longer WIP, but needs https://github.com/metal3-io/ironic-image/pull/410 to be merged first.

**Comment by tuminoid** [comment]
/cc @dtantsur @zaneb 

PTAL.

**Comment by tuminoid** [comment]
As requested on the ironic-image PR, I changed inspector container to run as ironic-inspector user, and keepalived back to nonroot as it is not sharing the filesystem with other containers.

**Comment by tuminoid** [comment]
Let's see if it picks up the new ironic-image now.

/test-centos-e2e-integration-main
/test-ubuntu-integration-main

**Comment by tuminoid** [comment]
OK, we're passing tests here and in dev-env after fixing CI flake (corrupted node image was built). All comments should be addressed.

Can I get LGTM and approve for this as well? @zaneb @dtantsur @elfosardo

File: resources/keepalived-docker/configure-nonroot.sh
Comment (kashifest) [inline resources/keepalived-docker/configure-nonroot.sh:11]:
```suggestion
# run as non-root, allow editing the keepalived.conf during startup
```

**Review by kashifest — APPROVED** [review]
Thanks for this work. One small nit inline. Otherwise approved.

**Comment by kashifest** [comment]
Ow it had lgtm already. Didnt notice.

File: resources/keepalived-docker/configure-nonroot.sh
Comment (tuminoid) [reply resources/keepalived-docker/configure-nonroot.sh:11]: I will fix that typo in some future PR.

**Pull request merged**
